libsolidity/formal/CHC.cpp

```diff
@@ -898,7 +898,7 @@ void CHC::defineInterfacesAndSummaries(SourceUnit const& _source)
 						createVariable(*var);
 					for (auto var: function->returnParameters())
 						createVariable(*var);
-					for (auto const* var: localVariablesIncludingModifiers(*function))
+					for (auto const* var: localVariablesIncludingModifiers(*function, base))
```

Comment:
I believe we want to use `base` here and not `contract`, as we want the variables for the function in the context of `base`.

Reply:
Not sure, I think it should be `contract` because we're looping over all base contracts of `contract` collecting all their functions as functions of `contract`. In that case, if `function` has virtual modifiers, they should be resolved in the context of `contract`.

Reply:
I think you are right, although, I could not find an example where it would break.